libsawtooth/src/client/mod.rs

```diff
@@ -0,0 +1,155 @@
+/*
+ * Copyright 2018 Intel Corporation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * ------------------------------------------------------------------------------
+ */
+
+mod error;
+
+use reqwest::{blocking::Client, Url};
+use serde::Deserialize;
+use std::collections::VecDeque;
+
+pub use error::SawtoothClientError;
+
+/// Client for Sawtooth Cli transactions
+/// Handles communication with the REST API
+pub struct RestApiSawtoothClient {
+    url: String,
+}
+
+trait SawtoothClient {
+    fn new(url: &str) -> Self;
```

Comment:
This shouldn't be the trait - URL in this case is specific to the REST implementation

Reply:
Fixed